securesystemslib/signer.py

```diff
@@ -29,6 +29,13 @@ def __init__(self, keyid: str, sig: str):
         self.signature = sig
 
 
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, Signature):
+            return False
+
+        return self.keyid == other.keyid and self.signature == other.signature
```

Comment:
This should likely handle the `other_headers` field used in GPG signatures. Also, should we be worried about different methods of key ID calculation? This may be an uneducated question on my part. :smile:

Reply:
I don't agree it should handle `other_headers` fied used in GPG signatures considering it doesn't have access to that field.
If we want to implement `__eq__` for the `GPGSignature` as well we will have to add it's own separate `__eq__` implementation looking like this:
```python
    def __eq__(self, other: Any) -> bool:
        if not isinstance(other, GPGSignature):
            return False

        return super().__eq__(other) and self.other_headers == other.other_headers

```

Reply:
The definition of `KEYID` is the following
```
The identifier of the key signing the ROLE object, which is a hexdigest of the
SHA-256 hash of the canonical form of the key. 
```
Meaning `keyid` will be a string in all cases and strings have defined `__eq__` methods.

Source: https://theupdateframework.github.io/specification/latest/#role-keyid

Reply:
> GPGSignature

I was confused because I didn't see it in the repo yesterday, but it's still an open PR. Makes sense to me, then. :)

> KEYID

Cool! I just wanted to explicitly clarify it, thank you!